### PR TITLE
Fix bugs found while writing tests for the lte link controller

### PR DESCRIPTION
**Pull request by maxd-nordic**
While looking through the link controller code for writing tests,
I encountered some bugs that should be fixed with these changes.

### CODE REVIEW

**Review by jhn-nordic — APPROVED** [review]

File: lib/lte_link_control/lte_lc.c [hunk 1]
```diff
@@ -1088,10 +1088,10 @@ int lte_lc_nw_reg_status_get(enum lte_lc_nw_reg_status *status)
 
 	if (!is_cellid_valid(cell_id)) {
 		*status = LTE_LC_NW_REG_UNKNOWN;
+	} else {
```
Comment (tokangas) [inline lib/lte_link_control/lte_lc.c:1091]: Isn't the commit message incorrect? Doesn't this fix affect the case when the cell ID is actually invalid? Earlier `LTE_LC_NW_REG_UNKNOWN` was overwritten by `status_tmp` when cell ID was not valid.

Comment (maxd-nordic) [reply lib/lte_link_control/lte_lc.c:1091]: This just adds another branch for `is_cellid_valid() == true`. I don't think it does anything if the cell ID is actually invalid. Where does this overwriting happen? Currently, we just discard `status_tmp`.

Comment (tokangas) [reply lib/lte_link_control/lte_lc.c:1091]: The original implementation was like this:

```
	if (!is_cellid_valid(cell_id)) {
		*status = LTE_LC_NW_REG_UNKNOWN;
	}

	*status = status_tmp;
```

So, the behavior was incorrect when cell ID was not valid. In that case `status` was first set to `LTE_LC_NW_REG_UNKNOWN`, but immediately overwritten by the value in `status_tmp` (which may have been different).

Comment (maxd-nordic) [reply lib/lte_link_control/lte_lc.c:1091]: Now I understand. Thanks for pointing that out. I'll update the commit message.

**Review by tokangas — APPROVED** [review]

**Review by simensrostad — APPROVED** [review]
Nothing suspicious. lgtm

**Review by gregersrygg — APPROVED** [review]
LGTM

**Pull request merged**